### PR TITLE
Modal: TS-migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3422179360

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  description = "",
+  onClose = NOOP,
+  alertDialog = false,
+  children,
+  triggerElement,
+  width = ModalWidth.DEFAULT,
+  hideCloseButton = false,
+  closeButtonAriaLabel = "close"
+}) => {
+  const childrenArray: ReactElement[] = useMemo(() => React.Children.toArray(children) as ReactElement[], [children]);
```
Comment (orrgottlieb) [inline src/components/Modal/Modal.tsx:84]: what happens if `children === null` ?

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      >
+        {header}
+        {content}
+        {footer}
+      </div>
+    </div>,
+    document.body
+  );
+
+  return <>{dialog}</>;
```
Comment (orrgottlieb) [inline src/components/Modal/Modal.tsx:156]: why do we need `<></>` can we change it to ```  return ReactDOM.createPortal(...```

File: src/components/Modal/useShowHideModal.ts [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
 
-export default function useShowHideModal({ instance, show, triggerElement, onClose, alertDialog }) {
+export default function useShowHideModal({
+  instance,
+  show,
+  triggerElement,
+  onClose,
+  alertDialog
+}: {
+  // the type is A11yDialog, but it's not exported, so can't be used
```
Comment (orrgottlieb) [inline src/components/Modal/useShowHideModal.ts:12]: lets create a `A11yDialogType` - have it set to any and once we have something better we could replace it in one place

File: src/components/ModalFooter/ModalFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { FC, ReactElement } from "react";
+import cx from "classnames";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import classes from "./ModalFooter.module.scss";
+
+interface ModalFooterProps extends VibeComponentProps {
+  children: ReactElement | ReactElement[] | string;
```
Comment (orrgottlieb) [inline src/components/ModalFooter/ModalFooter.tsx:7]: @SergeyRoyt Please add task to create a unified children type so would could use it everywhere?

File: src/components/ModalHeader/ModalHeader.tsx [hunk 5]
```diff
@@ -1,23 +1,75 @@
...
   title,
   titleClassName,
-  description,
+  description = "",
   descriptionClassName,
   icon,
-  closeModal,
-  iconSize,
+  closeModal = NOOP,
+  iconSize = 24,
```
Comment (orrgottlieb) [inline src/components/ModalHeader/ModalHeader.tsx:69]: Magic number 🪄

**Review by orrgottlieb — APPROVED** [review]

File: src/components/ModalFooter/ModalFooter.tsx (hunk 4 above)
Comment (SergeyRoyt) [reply src/components/ModalFooter/ModalFooter.tsx:7]: Yes, was thinking about that too. Will do

File: src/components/Modal/Modal.tsx (hunk 1 above)
Comment (SergeyRoyt) [reply src/components/Modal/Modal.tsx:84]: Fixed

File: src/components/Modal/useShowHideModal.ts (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/Modal/useShowHideModal.ts:12]: Done

File: src/components/ModalHeader/ModalHeader.tsx (hunk 5 above)
Comment (SergeyRoyt) [reply src/components/ModalHeader/ModalHeader.tsx:69]: 🪄 🧙

**Pull request merged**